cypress/integration/admin.spec.js

```diff
@@ -0,0 +1,44 @@
+const { join } = require('path');
+
+context('Admin interface', () => {
+  it('should redirect anonymous users to the login page', function() {
+    cy.visit('/admin')
+    cy.location().should('match', /\/admin\/login\//)
+  })
+
+  function login() {
```

Comment:
We might not have to use the user interface to log in for each test, which the Cypress docs treat as an anti-pattern. (Though I vaguely remember you saying that it was difficult to log in programmatically in the app? Just ignore this if so.) We could [add a `cy.login()` command](https://docs.cypress.io/api/cypress-api/custom-commands.html#Syntax) for testing purposes.

Cypress has some useful docs on logging in during tests:

* https://docs.cypress.io/guides/references/best-practices.html#Organizing-Tests-Logging-In-Controlling-State
* https://github.com/cypress-io/cypress-example-recipes#logging-in-recipes